crates/astria-composer/src/searcher/executor.rs

```diff
@@ -164,28 +216,43 @@ impl Executor {
                             break Err(e).wrap_err("failed submitting bundle to sequencer");
                         }
                     }
+
+                    block_timer.as_mut().reset(Instant::now() + Duration::from_millis(self.block_time_ms));
                 }
 
-                // receive new bundle for processing
-                Some(bundle) = self.new_bundles.recv(), if submission_fut.is_terminated() => {
-                    // TODO(https://github.com/astriaorg/astria/issues/476): Attach the hash of the
-                    // bundle to the span. Linked GH issue is for agreeing on a hash for `SignedTransaction`,
-                    // but both should be addressed.
-                    let span =  info_span!(
-                        "submit bundle",
-                        nonce.initial = nonce,
-                        bundle.len = bundle.len(),
+                Some(bundle) = bundle_factory.pop_complete(), if submission_fut.is_terminated() => {
```

Comment:
IMO remove this arm and merge this into  the "submission completed" branch - if the arm returns and there is a new bundle ready to be sent, schedule it immediately without taking an extra iteration through the loop first.

Reply:
I tried this but it results in the first bundle not being fired until the block timer hits, even if it is filled and a new one starts. for that matter, not matter how many `SequenceAction`s are pushed into the `BundleFactory` - they won't be fired until after the first timer tick.

This happens if the submission drivers are only submission responses and timer ticks. Since there won't be a submission response before anything is submitted, it will have to wait for the block timer to tick.

This behavior will also happen at any point where the `BundleFactory` is emptied out entirely:
1. The response for the last bundle submission will be returned
2. The `BundleFactory` will be checked for additional bundles (`pop_completed`), which will return `None`
3. This branch won't fire off another `submission_fut`
4. Loop will continue until next timer tick

Possible solutions:
1. Reducing the timer to less than half of a block time, to make sure it'll hit. This is an easy fix but feel finnicky?
2. Not caring about this and deciding it is acceptable behavior. Will add documentation to reflect this if we want to do this.
3. Current state of the code, and to my understanding this requires the interface on `BundleFactory.pop_completed()` to return a `futures::Ready` so that the `pop_completed` call can be put in the select loop.

This felt like the best solution. WDYT?